app/utils/ServiceWatcher.js

```diff
@@ -0,0 +1,104 @@
+const {
+  SUBSTRATE_STATUS_POLL_PERIOD_MS,
+  SUBSTRATE_STATUS_TIMEOUT_MS
+} = require('../env')
+
+class ServiceWatcher {
+  // taking helper functions can be improved by taking an object
+  // { name: <srv_name>, method: createNodeApi } and use in init()
+  constructor(createNodeApi) {
+    this.report = {}
+    this.pollPeriod = SUBSTRATE_STATUS_POLL_PERIOD_MS
+    this.timeout = SUBSTRATE_STATUS_TIMEOUT_MS
+    this.createNodeApi = createNodeApi
```

Comment:
Should all of these be public?

Reply:
to be fair they all can be private apart from report

Reply:
you might want to hide the `report` value behind a getter then to ensure that it is always interpretted as read only

Reply:
also updated:)